apps/web/src/lib/functions/file-loaders/epub/generate-epub-html.ts

```diff
@@ -139,32 +139,48 @@ export default function generateEpubHtml(
       htmlHref = itemIdToHtmlRef[itemIdRef];
     }
 
-    const regexResult = /.*<body(?:[^>]*id="(?<id>.+?)")*[^>]*>(?<body>(.|\s)+)<\/body>.*/.exec(
-      data[htmlHref] as string
-    )!;
+    let parsedContent = parser.parseFromString(data[htmlHref] as string, 'text/html');
+    let body = parsedContent.body;
 
-    const bodyId = regexResult?.groups?.id || '';
-    let innerHtml = regexResult?.groups?.body || '';
+    if (!body?.childElementCount) {
```

Comment:
one small change - text nodes are not elements so it would not work with body elements only having textnodes. so you you should check the length of child nodes instead (same for the check below)